core/Cargo.toml

```diff
@@ -380,6 +382,8 @@ prometheus-client = { version = "0.22.2", optional = true }
 tracing = { version = "0.1", optional = true }
 # for layers-dtrace
 probe = { version = "0.5.1", optional = true }
+rfc2047-decoder = { version = "1.0.5", optional = true }
+email-encoding = { version = "0.3.0", optional = true }
```

Comment:
Thank you very much for your effort. However, I discovered that even AWS's own Go SDK doesn't handle this: https://github.com/aws/aws-sdk-go-v2/blob/38fe812f7abf976b6b1a5122c0c3bf303502cfc5/service/s3/deserializers.go#L7167-L7175

I prefer to keep things simple since it's quite rare for users to include non-ASCII characters in metadata. We can also make it clear in our API that users should encode those values themselves.

Reply:
That sounds reasonable. Done.